bgpd/bgp_bmp.c

```diff
@@ -1755,6 +1756,18 @@ static void bmp_stats(struct event *thread)
 	}
 }
 
+static void bmp_mirror_limit_set_default(struct bgp *bgp)
+{
+	struct bmp_bgp *bmpbgp = bmp_bgp_find(bgp);
+
+	if (bmpbgp == NULL)
+		return;
+	if (bmpbgp->mirror_qsizelimit == ~0UL)
+		return;
+	bmpbgp->mirror_qsizelimit = ~0UL;
+	bmp_bgp_put(bmpbgp);
```

Comment:
so ... won't this deref?

Reply:
if you have a better design patter to apply to my use case, you are welcome:
I can try to apply it.

For the reason why I do this:
```
It is wishable to share the bmpbgp structure with a
separate BGP instance (lets say Y) that wants to import
the BMP information from X.
```

Reply:
That may be a reasonable question, I guess, but it's not a response to the review comment? The review comment is "this code is wrong".
The irony is that the general comment was "this implementation choice is hard to get right", and ... in fact it's not right, even in this first effort.